partition_snapshot_reader.hh

```diff
@@ -133,11 +159,16 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
             return e;
         }
 
-        const range_tombstone& pop_range_tombstone() {
+        range_tombstone pop_range_tombstone() {
```

Comment:
It's a little sad that we're forcing a copy here, but just a little.

Reply:
Maybe we can make the return type std::conditional_t?

Reply:
We are making a copy anyway at one point: from the in-memory representation -> reader's buffer. This is fine as long as we ensure we move this value from here on.

Reply:
It's not really copying, as the only usage of this function was in next_range_tombstone:
```
                    range_tombstone rt = pop_range_tombstone();
```
Even though it isn't guaranteed, I expect that the compiler would do copy elision here.